### PR TITLE
test: add details_page test

### PR DESCRIPTION
**Pull request by shan-shaji**

### CODE REVIEW

**Comment by Feichtmeier** [comment]
Thank you very much @shan-shaji :+1: 

I am not really good with building tests. So you will have to wait until jp finds time to review it, I hope this is okay for you? Best regards and thanks again

**Comment by shan-shaji** [comment]
Hey @shan-shaji, Yeah sure

File: test/widgets/details_page_test.dart [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:settings/view/widgets/detail_page.dart';
+import 'package:settings/view/widgets/menu_item.dart';
+import 'package:yaru_icons/widgets/yaru_icons.dart';
+import '../test_helper.dart';
+import 'package:flutter/material.dart';
+
+void main() {
+  group('- Details Page Test', () {
```
Comment (jpnurmi) [inline test/widgets/details_page_test.dart:13]: It's great to have logical groups, especially for things that share the same `setUp()` or `tearDown()`, but what's your motive for grouping the entire test into a group that has virtually the same name as the whole test does?

Top-level tests are essentially in a group already, which is named after the test file and shares a top-level `setUp()` and `tearDown()` too. As far as I can tell, this group only adds a seemingly redundant extra level to the tree of tests, or am I missing something?

1) run tests

    Screenshot from 2021-09-18 15-21-47

2) click to expand

    Screenshot from 2021-09-18 15-21-54

3) click to expand again

    Screenshot from 2021-09-18 15-21-58

Comment (jpnurmi) [reply test/widgets/details_page_test.dart:13]: Furthermore, I'd leave out the bullet point formatting of test names and let users have their tools visualize tests in their preferred ways depending on the context. You may have a tree in an IDE, or you may have a flat list on the command-line, for example. I'm not sure what is this format optimized for. :slightly_smiling_face:

File: test/widgets/details_page_test.dart [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    });
+
+    testWidgets(
+        '- If `tablet(context)==false` the appbar [leading] will contain [BackButton]',
+        (widgetTester) async {
+      const width = 414;
+      const height = 846;
+
+      widgetTester.binding.window.devicePixelRatioTestValue = (2.625);
+      widgetTester.binding.window.textScaleFactorTestValue = (1.1);
```
Comment (jpnurmi) [inline test/widgets/details_page_test.dart:57]: Where did these very specific magic values come from? Perhaps they are fine, I don't know, but I would be baffled if this test started suddenly failing. Perhaps a comment to explain what's going on here?

**Review by jpnurmi — COMMENTED** [review]
Thanks, I had only some little nit-pick and questions :slightly_smiling_face:

File: test/widgets/details_page_test.dart (hunk 1 above)
Comment (shan-shaji) [reply test/widgets/details_page_test.dart:13]: Hey @jpnurmi, Thank you for your valuable suggestions :+1: slightly_smiling_face. So I need to change the test descriptions and remove the unnecessary grouping.

File: test/widgets/details_page_test.dart (hunk 2 above)
Comment (shan-shaji) [reply test/widgets/details_page_test.dart]: Hey @jpnurmi, Could you please check the code. I will be happy to correct if there are any mistakes🙂.

**Pull request merged**